algorithms/src/snark/marlin/data_structures/proof.rs

```diff
@@ -26,6 +30,8 @@ use snarkvm_utilities::{
     ToBytes,
 };
 
+use std::collections::BTreeMap;
+
 #[derive(Clone, Debug, PartialEq, Eq, CanonicalSerialize, CanonicalDeserialize)]
 pub struct Commitments<E: PairingEngine> {
     pub witness_commitments: Vec<WitnessCommitments<E>>,
```

Comment:
Let's change this to `witness_commitments: Vec<Vec<WitnessCommitments<E>>>`, i.e. one inner `Vec` per circuit.